### PR TITLE
Adds snippet model and validations

### PR DESCRIPTION
**Pull request by jonathanly**
#### Snippet model has following attributes:
- user_id
- title
- body

#### Other things:
- belongs_to :user
- updates snippet factory

### CODE REVIEW

File: spec/models/snippet_spec.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require "rails_helper"
+
+RSpec.describe Snippet do
+  context "validations" do
+    it { is_expected.to validate_presence_of(:title) }
+    it { is_expected.to validate_presence_of(:body) }
+  end
+end
```
Comment (apalski) [inline spec/models/snippet_spec.rb]: Do you need to add the subject { build(:snippet) } to this spec, or was it passing ok?

File: db/migrate/20180112015947_create_snippets.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  def change
+    create_table :snippets do |t|
+      t.integer :user_id, foreign_key: true, index: true, null: false
+      t.string :title, null: false
+      t.text :body, null: false
+
+      t.timestamps
+    end
+  end
+end
```
Comment (apalski) [inline db/migrate/20180112015947_create_snippets.rb]: I like the way you've set the index up on the foreign key!

File: spec/models/snippet_spec.rb (hunk 1 above)
Comment (jonathanly) [reply spec/models/snippet_spec.rb]: It was passing okay.
Elle said assigning the subject is required only when you're testing for uniqueness.

Comment (apalski) [reply spec/models/snippet_spec.rb]: Cool, good to know!

**Comment by jonathanly** [comment]
Thanks Elle and Annette for reviewing the PR.
Merging into master now.

**Pull request merged**